net/redeemer.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package net;
+
+service TicketRedeemer {
+    rpc QueueTicket(Ticket) returns (google.protobuf.Empty) {}
+    rpc MaxFloat(MaxFloatRequest) returns (MaxFloatUpdate) {}
+    rpc MonitorMaxFloat(google.protobuf.Empty) returns (stream MaxFloatUpdate) {}
+}
+
+message MaxFloatRequest {
+    bytes sender = 1;
+}
+
+message Ticket {
```

Comment:
A `Payment` can only be for a single set of `TicketParams` thus only for a single `recipientRand` as well, right?  

I felt like duplicating the message definition makes the code quite a bit cleaner and nicer on the eye, easier type conversions and no loops. 

So the usefulness of reusing the `Payment` message to me seems to be related to whether we expect to receive multiple winning tickets within a single payment often. 

I can see the benefit of not having to send multiple RPC calls when we receive multiple winning tickets in a payment but do we really expect it to occur that often?

Reply:
> A Payment can only be for a single set of TicketParams thus only for a single recipientRand as well, right?

Ah yeah you're right. Only a single recipientRand would be needed for a payment that could contain multiple tickets.

> I felt like duplicating the message definition makes the code quite a bit cleaner and nicer on the eye, easier type conversions and no loops.

Yeah good point.

It might make sense to structure the Ticket message in a similar manner as the Payment message though i.e.

```golang
message Ticket {
    TicketParams ticket_params = 1;
    bytes sender = 2;
    TicketExpirationParams expiration_params = 3;
    TicketSenderParams sender_params = 4;
    bytes recipient_rand = 5;
}
```

This way if the ticket format ever happens to change both the Payment and Ticket messages can pick up those changes since they share the same structs i.e. TicketParams.

Reply:
Fixed in the following commits 

protofile , stubs & mocks: 258e02747160df22b465787adfd245ce9ed02a65
server: 28432f0acd05767f4310277263ef73bf63781e9f
client: 759114fb02f384681b8003e48efb2bb9801bd842